definitions/npm/react-redux_v5.x.x/flow_v0.68.0-/test_connect.js

```diff
@@ -57,6 +57,28 @@ function doesNotRequireDefinedComponentToTypeCheck1case() {
 }
 
 function doesNotRequireDefinedComponentToTypeCheck2case() {
+  type Props = {
+    stringProp: string,
+  };
+
+  const Component = ({ stringProp }: Props) => {
+    return <span>{stringProp}</span>;
+  };
+
+  // This is actually required to reproduce an issue with Flow and the `*` type.
+  function getBoolean() {
+    return false;
+  }
+
+  const mapStateToProps = (state: {}) => ({
+    // $ExpectError wrong type for numProp
+    numProp: getBoolean()
```

Comment:
Note that this case is almost the same as `doesNotRequireDefinedComponentToTypeCheck1case`, but the value from `numProp` comes from the `getBoolean` function, and we'd expect the same Flow error as `doesNotRequireDefinedComponentToTypeCheck1case` throws, but for some reason it is not throwing that.

Reply:
I remember that in some cases we need to "help" flow by adding `*` as return value for `mapStateToProps`, but I have no idea whether this was supposed to work better with the new bindings.

Reply:
No, `*` didn't work, thx for the suggestion!
I also have this try-flow link has a simplified version of `connect` with the exact same problem: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBLAtgBzgJwBdkwBDAZzACUBTUgY2KnzizAHJ87H2BudYMDABlbDhgYoGGgBMwYmDSw0AdoVKEMcFWDhQwXBoQC0XGQFcAHmAAG9bSpqMbYABIB5AMLpCATxw0YACypDjC6oQ0ACpwAAosOOQAPOEaNPFwiQB8YAC8YAAUAJR5OamRGYn8fgFuXkkA6vihATKerHiOalH+NAA0YACSKgBWTpEyleQ5+bRGAHTtuNqqhD0BSQAkACKSUElzjPMAoorKau0qUgDmjc04rUudq+s0WQPDY4yyU1lZ-FBzCpGFodPYVI5GElUGARLEAFxgdwAIy+hD6MLATRasieK26vURh0Iiw6+LWvSSIHeqCyBSwoXK0TiCXIiJCYQizKmKViWSKiIK2IeuLJXQpARKuRyHk8dxxbTFL16A2EfLAAG9MVxCOZ8DpAcDNNpCgh7gF8CUtbDYTq9WCYBRKMKLWAaJZIioZJRiaTluKkii0TkNQBffiw8OoUM+XpgKZ5TUAH0xpAAQnA4DBEcjM4pSCoMbDkRmszm83QVKgk1HUPRHeRKAANN0e1Te6jcEl4gO-TXR9CGkEm640QgAOXMWGRNHwxURKinM-w-dtY-tYAAzPwY6gh8adAzOWkYlNiquDOv9RfYYIwFEABY0ciBABMYAkjkopC4boAjuYGAAG6kIoagDMi5jEAAYvASAYJQKhwMQ9i4I6GAqBh1y6DoJZ5pisLpnmiKbkWNp4WWYCjhOS6zsUmJRru4LkMQACaibgpChD0oyXKnqyRQFE2RT8Ek7HAP8QA

Reply:
One thing that works is if I use `$Shape<Props>` as return, but I wanted to avoid having to do this, and I believe Flow should be able to properly detect that return value.

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBLAtgBzgJwBdkwBDAZzACUBTUgY2KnzizAHJ87H2BudYMDABlbDhgYoGGgBMwYmDSw0AdoVKEMcFWDhQwXBoQC0XGQFcAHmAAG9bSpqMbYABIB5AMLpCATxw0YACypDjC6oQ0ACpwAAosOOQAPOEaNPFwiQB8YAC8YAAUAJR5OamRGYn8fgFuXkkA6vihATKerHiOalH+NAA0YACSKgBWTpEyleQ5+bRGAHTtuNqqhD0BSQAkACKSUElzjPMAoorKau0qUgDmjc04rUudq+s0WQPDY4yyU1lZ-FBzCpGFodPYVI5GElUGARLEAFxgdwAIy+hD6MLATRasieK26vURh0Iiw6+LWvSSIHeqCyBSwoXK0TiCXIiJCYQizKmKViWSKiIK2IeuLJXQpARKuRyHk8dxxbTFL16A2EfLAAG9MVxCOZ8DpAcDNNpCgh7gF8CUtbDYTq9WCYBRKMKLWAaJZIioZJRiaTluKkii0TkNQBffiw8OoUM+XpgKZ5TUAH0xpAAQnA4DBEcjM4pSCoMbDkRmszm83QVKgk1HUPRHeRKAANN0e1Te6jcEl4gO-TXR9CGkEm640QgAOXMWGRNHwxURKinM-w-dtY-tYAAzPwY6gh8adAzOWkYlN52BNsIABahGhJPvWgzr-Wrm2CMBRK80ciBABMYAkRxKFILg3QAR3MDAADdSEUNQBmRcxiAAMXgJAMEoFQ4GIexcEdDAVAI65dB0Es80xWF0zzRFNyLG0yLLMBRwnJdZ2KTEo13cFyGIABNRNwUhQh6UZLlT1ZIoCibIp+CSfjgH+IA